src/js/select2/results.js

```diff
@@ -14,7 +14,7 @@ define([
 
   Results.prototype.render = function () {
     var $results = $(
-      '<ul class="select2-results__options" role="tree"></ul>'
+      '<ul class="select2-results__options" role="listbox"></ul>'
```

Comment:
A "tree" widget is something different, from the spec:
> A type of list that may contain sub-level nested groups _that can be collapsed and expanded_
The ARIA authoring practices describe the interaction screen readers should implement with ARIA widgets, I think using a Tree View is one of the main reasons the Select2 options are not announced. See version 1.0 and 1.1 (both working drafts):
https://www.w3.org/TR/wai-aria-practices/#TreeView
https://www.w3.org/TR/wai-aria-practices-1.1/#TreeView

On the other hand, the Combobox widget explicitly requires a listbox with options, see:
https://www.w3.org/TR/wai-aria-practices/#combobox
https://www.w3.org/TR/wai-aria-practices-1.1/#combobox

I'm afraid ARIA doesn't provide a way to emulate the semantics of `optgroup` or at least I wasn't able to find it. Nested groups would require further investigation and iteration. That's why I'm proposing to proceed with small steps and then iterate, if you agree :)

Testing the PR, only Firefox+NVDA announce the group "labels", IE11+JAWS skip the labels but do announce the options, Safari+VoiceOver announce just the "flat" ones.